### ISSUE TITLE
Implementation of Anvil's string splitting method is bad

### ISSUE THREAD

**Issue by DAK404** [DAK404/Truncheon#7]
The double quotes inside the string get removed.

For example:

mkdir "Hello "hw" World"

Is stored in arr as:

```
arr[0] = "mkdir"             //correct
arr[1] = "Hello hw World"    //expected Hello "hw" World
```

The rootcause in the code is found to be in the replaceAll() method used after the string splitting regex logic.

**Comment by DAK404** [comment]
Fix underway with a new logic to deal with strings enclosed within double quotes.

Logic prototype:

Replace the replaceAll() statement after splitting to check if the array index value starts and ends with a double quote, if true, then the first and last character is removed successfully.

If the first quote is not followed by an end quote, an error is returned since the array tries to get values from an empty array resulting in an ArrayOutOfBoundsException (which can be handled by a catch block again)

Else, the split string is returned, with no issues.

**Comment by DAK404** [comment]
Fixed the logic in the 231618482e933055fef289f88af323f07cc32c85 commit. Details can be seen here.
